conans/client/graph/graph_builder.py

```diff
@@ -410,11 +410,12 @@ def _resolve_recipe(self, current_node, dep_graph, requirement, check_updates,
             if not requirement.build_require:
                 raise ConanException(f"{current_node.ref} uses '<host_version>' in requires, "
                                      "it is only allowed in tool_requires")
-            transitive = current_node._public_deps.get(ref.name, context="host")
+            transitive = next((dep for dep in current_node.dependencies if
```

Comment:
Looks good enough, as it is under the ``if ref.version == "<host_version>":`` it cannot break anything that is not using ``host_version`` feature, and as long as it passes the old and the new test, lets move it forward.